### PR TITLE
COMMERCE-10598 Sanitize addresses for safe HTML viewing

### PR DESCRIPTION
**Pull request by brianikim**
From @victorcdp and @elocamp,

@andrea-ale-sbarra , this one is high priority. Security issue and also attached to LPP.

> Related issues: [COMMERCE-10598](https://issues.liferay.com/browse/COMMERCE-10598) @elocamp
> 
> The Address name and other fields were not being sanitized for HTML viewing. Proposed fix is to sanitize the values and labels.

### CODE REVIEW

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
@drewbrokke fyi

**Review by luca-pellizzon — APPROVED** [review]

**Review by andrea-ale-sbarra — APPROVED** [review]

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by victoroscuro** [comment]
ci:test:commerce

File: modules/apps/commerce/commerce-checkout-web/src/main/resources/META-INF/resources/checkout_step/address.jsp [hunk 1]
```diff
@@ -73,7 +73,7 @@ boolean hasManageAddressesPermission = baseAddressCheckoutStepDisplayContext.has
...
+						<aui:option data-city="<%= HtmlUtil.escapeAttribute(commerceAddress.getCity()) %>" data-country="<%= HtmlUtil.escapeAttribute(String.valueOf(commerceAddress.getCountryId())) %>" data-name="<%= HtmlUtil.escapeAttribute(commerceAddress.getName()) %>" data-phone-number="<%= HtmlUtil.escapeAttribute(commerceAddress.getPhoneNumber()) %>" data-region="<%= HtmlUtil.escapeAttribute(String.valueOf(commerceAddress.getRegionId())) %>" data-street-1="<%= HtmlUtil.escapeAttribute(commerceAddress.getStreet1()) %>" data-street-2="<%= Validator.isNotNull(commerceAddress.getStreet2()) ? HtmlUtil.escapeAttribute(commerceAddress.getStreet2()) : StringPool.BLANK %>" data-street-3="<%= Validator.isNotNull(commerceAddress.getStreet3()) ? HtmlUtil.escapeAttribute(commerceAddress.getStreet3()) : StringPool.BLANK %>" data-zip="<%= HtmlUtil.escapeAttribute(commerceAddress.getZip()) %>" label="<%= HtmlUtil.escape(commerceAddress.getName()) %>" selected="<%= selectedAddress %>" value="<%= commerceAddress.getCommerceAddressId() %>" />
```
Comment (drewbrokke) [inline modules/apps/commerce/commerce-checkout-web/src/main/resources/META-INF/resources/checkout_step/address.jsp:77]: I think it's not necessary to escape numeric values. You only need to worry about the strings.

**Review by drewbrokke — COMMENTED** [review]
Looks good, just the one inline comment.

File: modules/apps/commerce/commerce-checkout-web/src/main/resources/META-INF/resources/checkout_step/address.jsp (hunk 1 above)
Comment (brianikim) [reply modules/apps/commerce/commerce-checkout-web/src/main/resources/META-INF/resources/checkout_step/address.jsp:77]: @drewbrokke , only thing being changed is the label.

`label="<%= HtmlUtil.escape(commerceAddress.getName()) %>"`

Comment (drewbrokke) [reply modules/apps/commerce/commerce-checkout-web/src/main/resources/META-INF/resources/checkout_step/address.jsp:77]: Ah, got it, thank you for clarifying. Looks good then!

**Comment by baharturk** [comment]
ci:forward

**Pull request closed without merging**